### DISCUSSION TITLE
How to uninstall the installed local library

### DISCUSSION THREAD

**Discussion by SunilMallireddy** [pdm-project/pdm#1892] [Q&A]
Hi,

I have created a local library called `temp_lib` using pdm tool by running a command `pdm install`. The project library has `pyproject.toml` file. When I ran `pdm install`, it installed local package `temp_lib`. How to uninstall the local package `temp_lib`. I tried `pdm remove temp_lib`, but it fails as it checks for `temp_lib` in dependencies of `pyproject.toml` file. Could anyone help me on how to uninstall the package installed using pdm ?

**Comment by Matheus-Ribeiro-ITA** [comment]
Hi Sunil, I`m no PDM expert, but happy to try helping

For me on Mac, `pdm remove` is working ok:

```
(pdm_arm64) Matheus_Sampaio@* rb % pdm remove package_A
Removing packages from default dependencies: package_A
Changes are written to pyproject.toml.
🔒 Lock successful
Changes are written to pdm.lock.
Synchronizing working set with lock file: 0 to add, 1 to update, 8 to remove
```

Could you try adding the `temp_lib` to the pyproject.toml with `pdm add` and then remove it?
Also, could you share the error message here pls

**Reply by SunilMallireddy** [reply]
Hi Matheus,

I have `temp_lib` as a package using the command `pdm add temp_lib` and this is what I see as the output 
```Package temp_lib is the project itself.
Lock successful
Changes are written to pyproject.toml.
All packages are synced to date, nothing to do.
Installing the project as an editable package...
  ✔ Update temp_lib 0.1.0+editable -> 0.1.0 successful
All complete!```

When I tried to remove using the command `pdm remove temp_lib`, I am facing the same error.

```Removing packages from default dependencies: temp_lib
[ProjectError]: temp_lib does not exist in default dependencies.
```

I wanted to uninstall the project package itself. `temp_lib` is the project and I am trying to uninstall the temp_lib package.

**Reply by Matheus-Ribeiro-ITA** [reply]
Ohhh I see now, Sunil, you want to not install the project itself, right?

If so, try installing like this: `pdm install --no-self` on a new env. This will install all packages except the project itself.

I think you can also, after `pdm install --no-self`, uninstall the temp_lib with  `pip uninstall temp_lib`

Reference:
https://pdm.fming.dev/latest/reference/cli/#install

**Reply by SunilMallireddy** [reply]
I tried `pip uninstall temp_lib`, but its not able to find the package even though package exists.

**Reply by Matheus-Ribeiro-ITA** [reply]
I see. It will only work if you are installing on the virtual env, and you need to run on the virtual env the pip uninstall too. If you are using pep582, I don`t know much about it.

Take a look at `pdm install --no-self`, if don't work. I would recomend opening a feature request for `pdm remove project_itself`. Frost respond pretty quickly the issues

**Reply by SunilMallireddy** [reply]
I have raised the feature request and `pdm sync --no-self --clean` is for uninstalling the project. https://github.com/pdm-project/pdm/issues/1901#issuecomment-1541062866

**Comment by dranjan** [comment]
I expected `pdm sync --no-self --only-keep` would do this, but it actually doesn't remove the self-package, which surprised me. That seems like a reasonable feature to request IMHO. But as a workaround, @SunilMallireddy can you delete and then recreate the venv or `__pypackages__` directory with `--no-self`?

**Reply by SunilMallireddy** [reply]
I still have the same issue. The package is not uninstallable

**Reply by dranjan** [reply]
I think you're right. I was just saying that if you can avoid installing it in the first place (by using `--no-self` with PDM commands), then you wouldn't need to uninstall it. It's a potential workaround, not a solution.

**Comment by hitesh-pathak** [comment]
Is there any solution to this yet? Seems like once you get the library installed (in editable mode at least), there is simply no way to remove it.
Am I missing something, is this by design..?
